src/zcbenchmarks.cpp

```diff
@@ -322,3 +326,82 @@ double benchmark_increment_note_witnesses(size_t nTxs)
     return timer_stop(tv_start);
 }
 
+// Fake the input of a given block
+class FakeCoinsViewDB : public CCoinsViewDB {
+    uint256 hash;
+    ZCIncrementalMerkleTree t;
+
+public:
+    FakeCoinsViewDB(std::string dbName, uint256& hash) : CCoinsViewDB(dbName, 100, false, false), hash(hash) {}
+
+    bool GetAnchorAt(const uint256 &rt, ZCIncrementalMerkleTree &tree) const {
+        if (rt == t.root()) {
+            tree = t;
+            return true;
+        }
+        return false;
+    }
+
+    bool GetNullifier(const uint256 &nf) const {
```

Comment:
There is no need to fake it, because the block being benchmarked does not contain any JoinSplits. I specifically selected a block without them, for two reasons:

- Checking JoinSplits would require the benchmarks to have access to the mainnet parameters, which is significant additional development effort.
- The goal of this PR is to benchmark the slowest blocks we know about, which at this point are the blocks containing many-input transactions. A block completely full of JoinSplits would in theory be slower, but we have yet to see such a block, and the majority of the time cost is already captured in the existing `verifyjoinsplit` benchmark.